src/Haskell/Ide/Engine/LSP/Reactor.hs

```diff
@@ -17,19 +17,21 @@ where
 
 import           Control.Monad.Reader
 import qualified Data.Map                      as Map
-import qualified Language.Haskell.LSP.Core     as Core
-import qualified Language.Haskell.LSP.Messages as J
-import qualified Language.Haskell.LSP.Types    as J
 import           Haskell.Ide.Engine.Compat
 import           Haskell.Ide.Engine.Config
 import           Haskell.Ide.Engine.PluginsIdeMonads
 import qualified Haskell.Ide.Engine.Scheduler  as Scheduler
 import           Haskell.Ide.Engine.Types
+import qualified Language.Haskell.LSP.Core     as Core
+import qualified Language.Haskell.LSP.Messages as J
+import qualified Language.Haskell.LSP.Types    as J
+
+-- ---------------------------------------------------------------------
 
 data REnv = REnv
   { scheduler         :: Scheduler.Scheduler R
   , lspFuncs          :: Core.LspFuncs Config
-  , reactorPidCache   :: Int
+  , reactorPidCache   :: Int -- TODO:AZ: do we need this? what is it for?
```

Comment:
Sorry should have documented this. Since we need to make each LSP command ID unique, we use append the PID to it. It might be a micro-optimisation, but the PID is fetched once and then stored inside REnv and accessed with `getPidCache` on the `HasPidCache` class. Is this a bit overkill?

Reply:
Yes, I realised that, eventually. But, I grepped the code, and it is never used.  IIRC we simplified the way we do this, so don't need to keep this value around so much.